kolibri/utils/constants/installation_types.py

```diff
@@ -0,0 +1,24 @@
+"""
+This module contains constants representing the type of "installers" used to install Kolibri.
+"""
+from __future__ import unicode_literals
+
+APK = "apk"
+APT = "apt"
+KOLIBRI_SERVER = "kolibri-server"
+MACOS = "Mac"
+PEX = "pex"
+UWSGI = "uwsgi"
+WHL = "whl"
+WINDOWS = "Windows"
+
+install_type_map = {
+    APK: "APK",
+    APT: "APT - Debian package",
+    KOLIBRI_SERVER: "kolibri(apt) with kolibri_server",
+    MACOS: "MacOS Installer",
+    PEX: "PEX file",
+    UWSGI: "UWSGI process",
+    WHL: "WHL Python package",
+    WINDOWS: "Windows Installer",
+}
```

Comment:
I'm a bit confused why there are two inconsistent variations of these strings.

Also, if these are permanent strings that cannot be changed we might want to tighten them a bit and make them a bit more consistent and correct:

* package manager, file format, or operating system is the primary piece of info?
* "apt" is spelled two different ways above
* "MacOS" is actually spelled "macOS", and it's technically not an "installer", just an "app"
* For "Windows" we might want to distinguish the current "server" version from a potential future "app" version more similar to mac



```suggestion
ANDROID = "Android APK"
DEBIAN = "Debian package"
DEBIAN_APT = "Debian APT-installed package"
DEBIAN_APT_SERVER = "Debian APT package with kolibri-server"
MACOS_APP = "macOS desktop app"
PEX = "PEX executable"
UWSGI = "UWSGI process"
WHL = "WHL Python package"
WINDOWS_BG = "Windows background process"

runtime_types = (
    ANDROID,
    DEBIAN,
    DEBIAN_APT,
    DEBIAN_APT_SERVER,
    MACOS_APP,
    PEX,
    UWSGI,
    WHL,
    WINDOWS_BG,
)
```

Reply:
The spec for this is here: https://github.com/learningequality/kolibri/issues/7960 this does need some updates to fit that spec (such as being able to insert the version number), but better to iterate there. Also, as there is already historic data in this format for some installers, updates may not be preferred for consistent analytics reporting.

Reply:
Ah, didn't realize there there was already data. At the least then there should only be one set of strings, not two? And we can still choose more consistent constant names even if the actual strings are a bit noisy.

Reply:
@rtibbles great, I was not aware of #7960 happy to fix these inconsistences after knowing the details. 

@indirectlylit I agree with some of your changes (I didn't know we  have an app for MacOs instead of an installer) but not with others, For Windows, for example installation type is clearly an installer, not a background process.
Also , there's not practical way to know if a Debian package has been installed using `dpkg` or `apt`, the current logic tries to find it out, but it can be easily cheated. In my opinion, we should just say "Deb package" removing the `apt` reference.

Reply:
for what it's worth these comments are not critical or intended to be blocking.

> For Windows, for example installation type is clearly an installer, not a background process.

Is it the "installation type" we care about or the runtime app? Sometimes it's both but often they are independent. This is like the Debian package which can be installed either with `dpkg` or `apt`.

The APK can be installed by sideloading or via the app store.

The [current Windows repo](https://github.com/learningequality/kolibri-installer-windows) contains three things:
* an installer script which runs once
* a background process (`kolibri.exe`) which runs Kolibri in the background, potentially on start-up
* a small GUI which allows some basic config and control

Additionally at some point we'll also have a Windows "foreground app" similar to the Endless, Android, and macOS apps [based on this repo](https://github.com/learningequality/kolibri-app). When this happens, it may also have its own installer.

> In my opinion, we should just say "Deb package" removing the apt reference.

agreed!

Reply:
> Is it the "installation type" we care about or the runtime app? 

well, this is a part of a function called `installation_type` :smile: , probably knowing the runtime app is useful too, but when trying to get information from the user to solve issues, knowing how Kolibri was installed is important.